patroni/ha.py

```diff
@@ -554,14 +554,16 @@ def update_cluster_history(self):
         elif not cluster_history or cluster_history[-1][0] != master_timeline - 1 or len(cluster_history[-1]) != 4:
             cluster_history = {line[0]: line for line in cluster_history or []}
             history = self.state_handler.get_history(master_timeline)
+            history[-1].append(self.cluster.leader.name)
             if history and self.cluster.config:
                 history = history[-self.cluster.config.max_timelines_history:]
                 for line in history:
                     # enrich current history with promotion timestamps stored in DCS
                     if len(line) == 3 and line[0] in cluster_history \
-                            and len(cluster_history[line[0]]) == 4 \
+                            and len(cluster_history[line[0]]) == 5 \
                             and cluster_history[line[0]][1] == line[1]:
                         line.append(cluster_history[line[0]][3])
+                        line.append(cluster_history[line[0]][4])
```

Comment:
What if we are dealing with the "old" format and there are only 4 columns in a row?
We need to support it as well.

Example:
old history in DCS
```json
[
[1,1,"no recovery target specified", "current date"]
]
```

After that we upgrade and restart Patroni with pause enabled and it should be rewritten as:
```json
[
[1,1,"no recovery target specified", "current date", "my-name"]
]
```

Reply:
I have made changesmade change